webapp/javascript/pages/adhoc/components/FileUploader.tsx

```diff
@@ -1,21 +1,54 @@
-/* eslint-disable react/jsx-props-no-spreading */
-import React, { useCallback } from 'react';
+/* eslint-disable react/jsx-props-no-spreading, jsx-a11y/role-supports-aria-props */
+import React, { useCallback, useState } from 'react';
 import { useDropzone } from 'react-dropzone';
 import Button from '@webapp/ui/Button';
 import type { DropzoneOptions } from 'react-dropzone';
-
-// Note: I wanted to use https://fontawesome.com/v6.0/icons/arrow-up-from-bracket?s=solid
-// but it is in fontawesome v6 which is in beta and not released yet.
-import { faArrowAltCircleUp } from '@fortawesome/free-regular-svg-icons/faArrowAltCircleUp';
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faFileUpload } from '@fortawesome/free-solid-svg-icons/faFileUpload';
+import classNames from 'classnames/bind';
+import Dropdown, { MenuItem } from '@webapp/ui/Dropdown';
+import UploadIcon from './UploadIcon';
 import styles from './FileUploader.module.scss';
 
+const cx = classNames.bind(styles);
+
+const isJSONFile = (file: File) =>
+  file.name.match(/\.json$/) ||
+  file.type === 'application/json' ||
+  file.type === 'text/json';
+
+const spyNames = {
+  gospy: 'Go',
+  pyspy: 'Python',
+  phpspy: 'PHP',
+  'pyroscope-rs': 'Rust',
+  rbspy: 'Ruby',
+  javaspy: 'Java',
+  dotnetspy: '.NET',
+  nodespy: 'NodeJS',
+  ebpfspy: 'eBPF',
+  other: 'Other',
+};
+
+const availableUnits = {
```

Comment:
There are other units available. Let's pull that and `spynames` from https://github.com/pyroscope-io/pyroscope/blob/main/packages/pyroscope-models/src/spyName.ts and https://github.com/pyroscope-io/pyroscope/blob/main/packages/pyroscope-models/src/profile.ts#L14-L22